### PR TITLE
Introducing documentation for OCI Object Storage state store component

### LINKED ISSUE #2070: Provide documentation on OCI ObjectStorage State Store
**What content needs to be created or modified?**

There should be docs on the State Store implementation based on OCI ObjectStorage service that instruct application operators on how to configure state store components of type state.oci.objectstorage.

**Describe the solution you'd like**

Once the code for this component is (close to being) accepted, I will provide the documentation as prescribed in this repository.

**Where should the new material be placed?**

a new file called setup-oci-objectstorage.md should be created under daprdocs/content/en/reference/components-reference/supported-state-stores/.

It will be quite similar to setup-azure-tablestorage.md

**The associated pull request from dapr/dapr, dapr/components-contrib, or other Dapr code repos**

https://github.com/dapr/components-contrib/pull/1401

**Additional context**

### PR DESCRIPTION
**Pull request by lucasjellema**
Signed-off-by: lucasjellema <lucasjellema@gmail.com>

Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [X] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [X] [Read the contribution guide](https://docs.dapr.io/contributing/contributing-docs/)
- [X] Commands include options for Linux, MacOS, and Windows within codetabs
- [X] New file and folder names are globally unique
- [X] Page references use shortcodes instead of markdown or URL links
- [X] Images use HTML style and have alternative text
- [X] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

Introduction of documentation for the OCI Object Storage state store component. This component (in PR https://github.com/dapr/components-contrib/pull/1401) is accepted and merged into the master. 

This documentation is completely new. It follows the example of existing documentation (especially for Azure Table Storage).

### CODE REVIEW

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-oci-objectstorage.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+| as specified with **bucketName** in components.yaml | -  | nihilus | darth | category: dapr-state-store , expiry-time-from-ttl: 2022-01-06T08:34:32 
+
+The exact value of the expiry-time-from-ttl depends of course on the time at which the state was created and will be 120 seconds later than that moment.
+
+
+Note that expired state is not removed from the state store by this component. An application operator may decide to run a periodic job that does a form of garbage collection in order to explicitly remove all state that has an  **expiry-time-from-ttl** label with a timestamp in the past.
+
+## Concurrency
+
+OCI Object Storage state concurrency is achieved by using `ETag`s. Each object in OCI Object Storage is assigned a unique ETag when it is created or updated (aka replaced). When the Set and Delete requests for this state store specify the FirstWrite concurrency policy, then the request need to provide the actual ETag value for the state to be written or removed for the request to be successful.
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/setup-oci-objectstorage.md]:
```suggestion
OCI Object Storage state concurrency is achieved by using `ETag`s. Each object in OCI Object Storage is assigned a unique ETag when it is created or updated (aka replaced). When the `Set` and `Delete` requests for this state store specify the FirstWrite concurrency policy, then the request need to provide the actual ETag value for the state to be written or removed for the request to be successful. 
```

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-oci-objectstorage.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+If you wish to create the bucket for Dapr to use, you can do so beforehand. However, Object Storage state provider will create one - in the specified compartment - for you automatically if it doesn't exist.
+
+In order to setup OCI Object Storage as a state store, you will need the following properties:
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/setup-oci-objectstorage.md]:
```suggestion
In order to setup OCI Object Storage as a state store, you need the following properties:
```

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-oci-objectstorage.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+  -H "Content-Type: application/json"
+  -d '[
+        {
+          "key": "nihilus",
+          "value": "darth"
+        }
+      ]'
+```
+
+will create the following object:
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/setup-oci-objectstorage.md]:
```suggestion
creates the following object:
```

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-oci-objectstorage.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+  -d '[
+        {
+          "key": "temporary",
+          "value": "ephemeral",
+          "metadata": {"ttlInSeconds": "120"}}
+        }
+      ]'
+```
+
+will create the following object:
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/setup-oci-objectstorage.md]:
```suggestion
creates the following object:
```

**Review by msfussell — COMMENTED** [review]
Overall looks great. Did some light editing on this

**Comment by lucasjellema** [comment]
> Overall looks great. Did some light editing on this

thanks @msfussell for the review and your comment. I have accepted all proposed changes and committed them.

**Review by greenie-msft — APPROVED** [review]

**Pull request merged**